python/rascaline/rascaline/systems/pyscf.py

```diff
@@ -0,0 +1,116 @@
+            rawsymb = rawsymb[5:]  # Remove the prefix GHOST
+            return pyscf.data.elements._ELEMENTS_UPPER[rawsymb]
+        else:
+            raise RuntimeError("Unsupported atom symbol %s" % symb_or_chg)
+    else:
+        return pyscf.data.elements.ELEMENTS[symb_or_chg]
+
+
+class PyscfSystem(SystemBase):
+    """Implements :py:class:`rascaline.SystemBase` wrapping a
+    `pyscf.gto.mole.Mole`_ or `pyscf.pbc.gto.cell.Cell`_.
+
+    Since pyscf does not offer a neighbors list, this
+    implementation of system can only be used with
+    ``use_native_system=True`` in
+    :py:func:`rascaline.calculators.CalculatorBase.compute`.
+
+    Atomic species are assigned as the atomic number if the atom ``type`` is
+    one of the periodic table elements; or their opposite if they are
+    ghost atoms.
+    (Pyscf does not seem to support anything else)
+
+    Please note that while pyscf uses Bohrs as length units internally,
+    we convert those back into Angströms for rascaline.
+    A pyscf object's "unit" attribute determines the units of the coordinates
+    given *to pyscf*, which are by default angströms.
+
+    .. _pyscf.gto.mole.Mole: https://pyscf.org/user/gto.html
+    .. _pyscf.pbc.gto.cell.Cell: https://pyscf.org/user/pbc/gto.html
+    """
+
+    @staticmethod
+    def can_wrap(o):
+        # assumption: if we have a periodic system, then pyscf.pbc is defined
+        if hasattr(pyscf, "pbc"):
+            return isinstance(o, (pyscf.gto.mole.Mole, pyscf.pbc.gto.cell.Cell))
+        else:
+            return isinstance(o, pyscf.gto.mole.Mole)
+
+    def __init__(self, frame):
+        """
+        :param frame : `chemfiles.Frame`_ object object to be wrapped
+            in this ``ChemfilesSystem``
+        """
+        super().__init__()
+        if not self.can_wrap(frame):
+            raise Exception(
+                "this class expects pyscf.gto.mole.Mole"
+                + "or pyscf.pbc.gto.cell.Cell objects"
+            )
+
+        self._frame = frame
+        self._species = self._frame.atom_charges().copy()  # dtype=int32
+        for atm_i, species in enumerate(self._species):
+            if species == 0:
+                symb = self._frame.atom_symbol(atm_i)
+                chg = pyscf.data.elements.index(symb)
+                self._species[atm_i] = -chg
+        self.is_periodic = isinstance(self._frame, pyscf.pbc.gto.cell.Cell)
+
+    def size(self):
+        return self._frame.natm
+
+    def species(self):
+        return self._species
+
+    def positions(self):
+        return pyscf.data.nist.BOHR * self._frame.atom_coords()
+
+    def cell(self):
+        if self.is_periodic:
+            cell = self._frame.a
+            if self._frame.unit[0].lower() == "a":
+                # assume angströms, we are good to go
+                return cell
+            else:
+                # assume bohrs, correct this
+                return pyscf.data.nist.BOHS * np.asarray(cell)
```

Comment:
Should there be similar logic for positions?

Reply:
no, the positions are obtained from pyscf's internal representation (directly in Bohrs) while the cell is as-given-to-pyscf, so its unit can be either Bohrs or Angströms.

Reply:
that's a weird API from PySCF, but ok ^_^

Reply:
well, to be fair, I am not using "the proper API" to fetch the boundary information, but I don't think there *is* a proper API for this specific information.
...though there is mol._h which is always in Bohrs, and is the transposed version of what we want.

Reply:
ok, make sense!